dagfactory/dagbuilder.py

```
@@ -171,8 +171,13 @@ def get_dag_params(self) -> Dict[str, Any]:
         # If there are no default_args, add an empty dictionary
         dag_params["default_args"] = {} if "default_args" not in dag_params else dag_params["default_args"]
 
-        if utils.check_dict_key(dag_params, "schedule_interval") and dag_params["schedule_interval"] == "None":
-            dag_params["schedule_interval"] = None
+        # Convert from 'schedule_interval: str' to 'schedule: str'
+        if utils.check_dict_key(dag_params, "schedule_interval"):
+            if dag_params["schedule_interval"] == "None":
+                dag_params["schedule"] = None
+            else:
+                dag_params["schedule"] = dag_params["schedule_interval"]
+            del dag_params["schedule_interval"]
```

I'm not sure if we should silently convert `schedule_interval` to `schedule`. Maybe in case of `schedule_interval` supplied, we should error out, or at least we should log this conversion.

---

Since we're dropping support for airflow < 2.4 (including in our testing environment), maybe we should removed all `schedule_interval` handling and replaced it with a hard error.

I'll fix the tests first, and waiting for the reply : D